src/backend/optimizer/plan/planshare.c

```diff
@@ -211,6 +211,14 @@ prepare_plan_for_sharing(PlannerInfo *root, Plan *common)
 		Material *m = make_material(common);
 		shared = (Plan *) m;
 
+		if (!shared->targetlist &&
```

Comment:
These conditions do not correlate?
If common is ModifyTable then may shared->targetlist be not null?
May be leave only check if common is ModifyTable?